java/src/jni/h4jni.h

```diff
@@ -17,9 +17,10 @@
  *
  */
 
-#include <jni.h>
 #include <string.h>
 
+#include <jni.h>
+
```

Comment:
I reordered the headers so they are in OS/C --> JNI --> Local order and put them outside the extern C block